components/TextFields/src/MDCTextInputControllerDefault.m

```diff
@@ -493,7 +493,8 @@ - (void)movePlaceholderToUp:(BOOL)isToUp {
       self.textInput.placeholderLabel.transform = CGAffineTransformIdentity;
 
       self.textInput.placeholderLabel.textColor =
-          self.previousPlaceholderColor ?: self.inlinePlaceholderColor;
+          self.previousPlaceholderColor
+            ? self.previousPlaceholderColor : self.inlinePlaceholderColor;
```

Comment:
I'm beginning to question the utility of the ternary operator without an implicit second argument.  Once you end up on multiple lines it gets hard to parse.

Consider:
if (self.previousPlaceholderColor) {
  self.textInput.placeholderLabel.textColor = self.previousPlaceholderColor;
} else {
  self.textInput.placeholderLabel.textColor = self.inlinePlaceholderColor;
}